hnn_core/extracellular.py

```diff
@@ -0,0 +1,601 @@
+"""
+Handler classes to calculate extracellular electric potentials, such as the
+Local Field Potential (LFP), at ideal (point-like) electrodes based on net
+transmembrane currents of all neurons in the network.
+
+The code is inspired by [1], but important modifications were made to comply
+with the original derivation [2] of the 'line source approximation method'.
+
+References
+----------
+1 . Parasuram H, Nair B, D'Angelo E, Hines M, Naldi G, Diwakar S (2016)
+Computational Modeling of Single Neuron Extracellular Electric Potentials and
+Network Local Field Potentials using LFPsim. Front Comput Neurosci 10:65.
+2. Holt, G. R. (1998) A critical reexamination of some assumptions and
+   implications of cable theory in neurobiology. CalTech, PhD Thesis.
+"""
+
+# Authors: Mainak Jas <mjas@mgh.harvard.edu>
+#          Nick Tolley <nicholas_tolley@brown.edu>
+#          Sam Neymotin <samnemo@gmail.com>
+#          Christopher Bailey <cjb@cfin.au.dk>
+
+import numpy as np
+from copy import deepcopy
+from numpy.linalg import norm
+from neuron import h
+
+from .externals.mne import _validate_type, _check_option
+
+
+def _transfer_resistance(section, electrode_pos, conductivity, method,
+                         min_distance=0.5):
+    """Transfer resistance between section and electrode position.
+
+    To arrive at the extracellular potential, the value returned by this
+    function is multiplied by the net transmembrane current flowing through all
+    segments of the section. Hence the term "resistance" (voltage equals
+    current times resistance).
+
+    Parameters
+    ----------
+    section : h.Section()
+        The NEURON section.
+    electrode_pos : list (x, y, z)
+        The x, y, z coordinates of the electrode (in um)
+    conductivity : float
+        Extracellular conductivity (in S/m)
+    method : str
+        Approximation to use. ``'psa'`` (point source approximation) treats
+        each segment junction as a point extracellular current source.
+        ``'lsa'`` (line source approximation) treats each segment as a line
+        source of current, which extends from the previous to the next segment
+        center point: |---x---|, where x is the current segment flanked by |.
```

Comment:
what is `|` in this diagram?

Reply:
The 'x' is an element of `seg_ctr`, and the `|` are `start` and `end` on L118-9.